app/models/footprint.rb

```diff
@@ -4,4 +4,18 @@ class Footprint < ApplicationRecord
   validates :carbon_in_kg,
             :offset_cost_total,
             :offset_cost_currency, presence: true
+
+  def self.aggregate_footprint_for_month(user_id, month, year)
+    #If we add another service(flight?)
+    #it would have to be added in the the second where method if the flight
+    #table also has a carbon_in_kg column if not this active record
+    #call may have to be redone
+
+    aggregate_footprint = Footprint.joins(car_monthly_mileage: :car)
+    .where("LOWER(month) = ?", "#{month.downcase}")
+    .where(car_monthly_mileages: {year: year}, cars: {user_id: user_id})
+    .sum('carbon_in_kg')
```

Comment:
Sorry for leading you through a wild goose chase. So according to the miro board, the data front end needs an array of footprints for each month of the year. This is why we need the sum(carbon_in_kg) in the select statement and a group(:month). Currently, what you have will only return data for one month. Can you change the query to this: 

`Footprint.joins(car_monthly_mileage:  :car).select('car_monthly_mileages.month, sum(carbon_in_kg) as footprint').where('cars.user_id = ? AND car_monthly_mileages.year = ?', "#{user_id}", "#{year}").group(:month)`

Then, can you please update the tests to include multiple months (not just July)? Then, we will be able to see that the method aggregates correctly by month.